sdk/identity/azure-identity/src/managed_identity_source.cpp

```diff
@@ -250,20 +250,28 @@ std::unique_ptr<ManagedIdentitySource> AppServiceV2019ManagedIdentitySource::Cre
       credName, clientId, resourceId, options, "IDENTITY_ENDPOINT", "IDENTITY_HEADER", "2019");
 }
 
-// Cloud Shell doesn't support user-assigned managed identities
 std::unique_ptr<ManagedIdentitySource> CloudShellManagedIdentitySource::Create(
     std::string const& credName,
     std::string const& clientId,
-    std::string const&,
+    std::string const& resourceId,
```

Comment:
Should this be an `Azure::Core::ResourceIdentifier`?

Reply:
It doesn't need to be, no. The public surface area accepts `ResourceIdentifier`, but the endpoint requires a string to be passed-in as a query parameter. We fetch the string from the input and pass that along.

Reply:
To confirm: CloudShellManagedIdentitySource::Create is *not* a public API?

Reply:
Correct, it is **not**.